tests/php/Core/EnvironmentTest.php

```diff
@@ -66,4 +67,115 @@ public function testGetVariables()
         $this->assertEquals('fail', $vars['test']);
         $this->assertEquals('global', $GLOBALS['test']);
     }
+
+    public function provideHasEnv()
+    {
+        return [
+            [
+                'setAs' => '.env',
+                'value' => true,
+                'expected' => true,
+            ],
+            [
+                'setAs' => '.env',
+                'value' => false,
+                'expected' => true,
+            ],
+            [
+                'setAs' => '.env',
+                'value' => null,
+                'expected' => true,
+            ],
+            [
+                'setAs' => '_ENV',
+                'value' => true,
+                'expected' => true,
+            ],
+            [
+                'setAs' => '_ENV',
+                'value' => false,
+                'expected' => true,
+            ],
+            [
+                'setAs' => '_ENV',
+                'value' => null,
+                'expected' => true,
+            ],
+            [
+                'setAs' => '_SERVER',
+                'value' => true,
+                'expected' => true,
+            ],
+            [
+                'setAs' => '_SERVER',
+                'value' => false,
+                'expected' => true,
+            ],
+            [
+                'setAs' => '_SERVER',
+                'value' => null,
+                'expected' => true,
+            ],
+            [
+                'setAs' => 'putenv',
+                'value' => true,
+                'expected' => true,
+            ],
+            [
+                'setAs' => 'putenv',
+                'value' => false,
+                'expected' => true,
+            ],
+            [
+                'setAs' => 'putenv',
+                'value' => null,
+                'expected' => true,
+            ],
+            [
+                'setAs' => null,
+                'value' => null,
+                'expected' => false,
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider provideHasEnv
+     */
+    public function testHasEnv(?string $setAs, ?bool $value, bool $expected)
+    {
+        $name = '_ENVTEST_HAS_ENV';
+
+        // Set the value
+        switch ($setAs) {
+            case '.env':
+                Environment::setEnv($name, $value);
+                break;
+            case '_ENV':
+                $_ENV[$name] = $value;
+                break;
+            case '_SERVER':
+                $_SERVER[$name] = $value;
+                break;
+            case 'putenv':
+                if ($value === null) {
+                    $val = 'null';
+                } else {
+                    $val = $value ? 'true' : 'false';
+                }
+                putenv("$name=$val");
+                break;
+            // null is no-op, to validate not setting it works as expected.
```

Comment:
```suggestion
            default:
                // setting null is no-op, to validate not setting it works as expected.
                break;
```

Reply:
Done. Didn't add `break` there because `break` after `default` is an anti-pattern. Also throwing an exception if the value here isn't `null` because that means there's something wrong with the test case.